lib/flutter_chips_selector.dart

```diff
@@ -199,120 +212,109 @@ class ChipsSelectorState<T> extends State<ChipsSelector<T?>> {
         child: FocusableActionDetector(
           focusNode: _focusableActionDetectorFocusNode,
           shortcuts: {
-            LogicalKeySet(LogicalKeyboardKey.arrowDown):
-                DirectionalFocusIntent(TraversalDirection.down),
-            LogicalKeySet(LogicalKeyboardKey.arrowUp):
-                DirectionalFocusIntent(TraversalDirection.up),
+            LogicalKeySet(LogicalKeyboardKey.arrowDown): _TraversalDownFocusIntent(),
+            LogicalKeySet(LogicalKeyboardKey.arrowUp): _TraversalUpFocusIntent(),
             LogicalKeySet(LogicalKeyboardKey.enter): SelectIntent(),
+            LogicalKeySet(LogicalKeyboardKey.delete): _RemoveLastIntent(),
+            LogicalKeySet(LogicalKeyboardKey.backspace): _RemoveLastIntent(),
           },
           actions: {
-            DirectionalFocusIntent: CallbackAction<DirectionalFocusIntent>(
-              onInvoke: (intent) {
-                if (intent.direction == TraversalDirection.down) {
-                  if (_suggestions.length > 0) {
-                    setState(
-                      () {
-                        _selectedIndex =
-                            (_selectedIndex + 1 >= _suggestions.length)
-                                ? 0
-                                : _selectedIndex + 1;
-                        _overlayEntry.markNeedsBuild();
-                      },
-                    );
-                    _scrollDown();
-                  }
-                } else if (intent.direction == TraversalDirection.up) {
-                  if (_suggestions.length > 0) {
-                    setState(
-                      () {
-                        _selectedIndex = (_selectedIndex - 1 < 0)
-                            ? _suggestions.length - 1
-                            : _selectedIndex - 1;
-                        _overlayEntry.markNeedsBuild();
-                      },
-                    );
-                    _scrollUp();
-                  }
-                }
-                return;
-              },
-            ),
-            SelectIntent: CallbackAction<SelectIntent>(onInvoke: (_) {
-              if (_selectedIndex > -1) {
-                selectSuggestion(_suggestions[_selectedIndex]);
-                setState(() {
-                  _selectedIndex = -1;
-                });
-              }
-              return;
-            }),
-          },
-          child: RawKeyboardListener(
-            focusNode: _rawKeyboardListenerFocusNode,
-            onKey: (RawKeyEvent event) {
-              if (event.isKeyPressed(LogicalKeyboardKey.delete) ||
-                  event.isKeyPressed(LogicalKeyboardKey.backspace)) {
-                if (_textController.text.length == 0 && _items.length > 0) {
+            if (_textController.text.length == 0 && _items.length > 0)
```

Comment:
I like it, but we would need to listen to text controller selection change and rebuild on those.

We do setState on text changes via the TextField, so deleting the first char would give us what we want.

But i think since this widget is rather limited regarding interactions with previously created chips, id say we leave it as is

Reply:
I'm glad we still tried 🤝